### PR TITLE
Buy view

### PR DESCRIPTION
**Pull request by Z-Ramezani**

### CODE REVIEW

File: Enigma/Group/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        ),
+        migrations.CreateModel(
+            name='members',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('groupID', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='group', to='Group.group')),
+                ('memmberID', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='memmber', to=settings.AUTH_USER_MODEL)),
+            ],
+        ),
+    ]
```
Comment (Shafizadegan) [inline Enigma/Group/migrations/0001_initial.py:33]: ok

File: Enigma/Group/models.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+    currency = models.CharField(max_length=100, default="تومان")
 
 
 class members(models.Model):
-    groupID = models.ForeignKey(Group, related_name='members', on_delete=models.CASCADE)
-    
+    groupID = models.ForeignKey(
+        Group, related_name='group', on_delete=models.CASCADE)
+    memmberID = models.ForeignKey(
+        MyUser, related_name="memmber", on_delete=models.CASCADE)
```
Comment (Shafizadegan) [inline Enigma/Group/models.py]: userID is not available in database

File: Enigma/buy/models.py [hunk 3]
```diff
@@ -18,13 +18,24 @@ class buyer(models.Model):
...
+    percent = models.IntegerField()
+
+
+class consum_contrib(models.Model):
+    contrib = models.ForeignKey(consumer, on_delete=models.CASCADE)
+    percent = models.IntegerField()
+
+
+class buyer_contrib(models.Model):
+    contrib = models.ForeignKey(buyer, on_delete=models.CASCADE)
```
Comment (Shafizadegan) [inline Enigma/buy/models.py]: edit contribution

File: Enigma/buy/models.py [hunk 4]
```diff
@@ -28,7 +27,6 @@ class consumer(models.Model):
         buy, related_name='consumers', on_delete=models.CASCADE)
     userID = models.ForeignKey(
         MyUser, related_name='userID_cons', on_delete=models.CASCADE)
-    percent = models.IntegerField()
```
Comment (Shafizadegan) [inline Enigma/buy/models.py]: ok

File: Enigma/buy/models.py [hunk 5]
```diff
@@ -30,10 +32,10 @@ class consumer(models.Model):
 
 
 class consum_contrib(models.Model):
-    contrib = models.ForeignKey(consumer, on_delete=models.CASCADE)
-    percent = models.IntegerField()
+    consumeID = models.ForeignKey(consumer, on_delete=models.CASCADE)
```
Comment (Shafizadegan) [inline Enigma/buy/models.py]: ok

File: Enigma/buy/views.py [hunk 6]
```diff
@@ -1,3 +1,20 @@
...
+
+    def all_group_buys(self, request):
+        perch = buy.objects.filter( groupID=request.data['groupID'] ).values()
+        perchase = buySerializer(instance=perch, many=True)
+        return response(perchase)
+
+    def user_group_buys(self, request):
+        perch = buy.objects.filter( groupID=request.data['groupID'] , userID=request.data['userID'] ).values()
+        perchase = buySerializer(instance=perch, many=True)
+        return response(perchase)
```
Comment (Shafizadegan) [inline Enigma/buy/views.py:36]: ok

**Review by Z-Ramezani — COMMENTED** [review]
:) all views work.

**Comment by fatemeshafiee** [comment]
Seems ok!

**Pull request merged**